pkg/detectors/azurestorage/azurestorage.go

```diff
@@ -98,6 +111,19 @@ func verifyAzureStorageKey(ctx context.Context, client *http.Client, accountName
 	}
 	defer res.Body.Close()
 
+	// parse container names and append in extra data
+	body, err := io.ReadAll(res.Body)
+	if err == nil {
+		response := storageResponse{}
+		if err := xml.Unmarshal(body, &response); err == nil {
+			var containerNames []string
+			for _, c := range response.Containers.Container {
+				containerNames = append(containerNames, c.Name)
+			}
+			extraData["container_names"] = strings.Join(containerNames, ", ")
+		}
+	}
```

Comment:
This should probably be inside the `http.StatusOk` block, as I doubt it would return the info for non-success responses.

Reply:
make sense. I have moved the parsing and unmarshaling code in StatusOk block.